tests/core/test_logging_config.py

```diff
@@ -319,9 +322,10 @@ def test_log_group_arns_remote_logging_with_cloudwatch_handler(
 
     def test_loading_remote_logging_with_kwargs(self):
         """Test if logging can be configured successfully with kwargs"""
+        pytest.importorskip("airflow.providers.amazon", reason="'amazon' provider not installed")
         from airflow.config_templates import airflow_local_settings
         from airflow.logging_config import configure_logging
-        from airflow.utils.log.s3_task_handler import S3TaskHandler
+        from airflow.providers.amazon.aws.log.s3_task_handler import S3TaskHandler
```

Comment:
I'm curious if these tests should sit in core or in providers?
and if they should be in core.. maybe we need a way to make sure that each new/existing logging integration adds such test?

Reply:
Yes, Moving those tests to providers is good idea. There is a long-standing issue of making sure that we do not refer from airflow core (including tests) to providers.

Reply:
Remote Logging is coupled with providers in core, at least with default template.
Need to think how to improve remote logging and decouple remote logging from the core to entirely into the providers and keep backward compatibility

Reply:
Yes. That's main reasons it's not done yet - while we managed to move configuration and few other things, this one remains coupled with the core.